### PR TITLE
BUG: Extra space is inserted on first line for long elements

### PR DESCRIPTION
**Pull request by eric-wieser**
Fixes gh-10181

Also strip some trailing spaces added by the web editor in the release notes.

### CODE REVIEW

File: doc/release/1.14.0-notes.rst [hunk 1]
```diff
@@ -265,26 +265,26 @@ In summary, the major changes are:
...
+
 * For line-wrapping:
 
   * The "dtype" part of ndarray reprs will now be printed on the next line
     if there isn't space on the last line of array output.
   * The ``linewidth`` format option is now always respected.
     The `repr` or `str` of an array will never exceed this, unless a single
     element is too wide.
-  * All but the last line of array strings will contain the same number of
-    elements.
```
Comment (eric-wieser) [inline doc/release/1.14.0-notes.rst]: This is only a change in extremely contrived scenarios (custom separators), so probably doesn't belong here.

**Comment by charris** [comment]
LGTM.

File: numpy/core/arrayprint.py [hunk 2]
```diff
@@ -622,8 +622,13 @@ def array2string(a, max_line_width=None, precision=None,
     return _array2string(a, options, separator, prefix)
 
 
-def _extendLine(s, line, word, line_width, next_line_prefix):
-    if len(line) + len(word) > line_width:
+def _extendLine(s, line, word, line_width, next_line_prefix, legacy):
+    needs_wrap = len(line) + len(word) > line_width
+    if legacy != '1.13' and len(line) == len(next_line_prefix):
```
Comment (eric-wieser) [inline numpy/core/arrayprint.py]: I suppose strictly this should be `<=`, but in practice that never happens

Comment (charris) [reply numpy/core/arrayprint.py]: Sure you don't want to change it before this goes in?

Comment (eric-wieser) [reply numpy/core/arrayprint.py]: Guess I will. Updated

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**